### PR TITLE
Add remaining_time to SlLoS.update

### LINKED ISSUE #2219: Change backend for enhance time edit menu
Make appropriate changes for OpenSlides/openslides-client#3296

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #2219

### CODE REVIEW

File: tests/system/action/structure_level_list_of_speakers/test_update.py [hunk 1]
```diff
@@ -138,6 +138,27 @@ def test_set_initial_time_with_other_field(self) -> None:
...
+        self.assertIn(
+            "Cannot set remaining_time and spoken_time at the same time.",
+            response.json["message"],
+        )
+
+    def test_set_remaining_time(self) -> None:
+        response = self.request(
+            "structure_level_list_of_speakers.update",
+            {"id": 3, "remaining_time": 700},
+            internal=True,
```
Comment (jsangmeister) [inline tests/system/action/structure_level_list_of_speakers/test_update.py:156]: Please remove the `internal=True`, this should be callable from the client

**Review by jsangmeister — CHANGES_REQUESTED** [review]

**Review by jsangmeister — CHANGES_REQUESTED** [review]
If the `remaining_time` is updated while `current_start_time` is not `None` (meaning some speaker is currently speaking for the structure level), the difference between the old and the new remaining time must be added/substracted from the `current_start_time`

File: openslides_backend/action/actions/structure_level_list_of_speakers/update.py [hunk 2]
```diff
@@ -36,10 +36,19 @@ def update_instance(self, instance: dict[str, Any]) -> dict[str, Any]:
                         f"Cannot set initial_time and {field} at the same time."
                     )
             instance["remaining_time"] = instance["initial_time"]
```
Comment (jsangmeister) [inline openslides_backend/action/actions/structure_level_list_of_speakers/update.py:38]: `current_start_time` also needs to be changed if `remaining_time` is changed by changing the `initial_time`

File: openslides_backend/action/actions/structure_level_list_of_speakers/update.py [hunk 3]
```diff
@@ -36,10 +36,19 @@ def update_instance(self, instance: dict[str, Any]) -> dict[str, Any]:
...
-                "Cannot set remaining_time and spoken_time at the same time."
+        elif "remaining_time" in instance:
+            if "spoken_time" in instance:
+                raise ActionException(
+                    "Cannot set remaining_time and spoken_time at the same time."
+                )
+            db_instance = self.datastore.get(
+                fqid_from_collection_and_id(self.model.collection, instance["id"]),
+                ["current_start_time", "remaining_time"],
             )
```
Comment (jsangmeister) [inline openslides_backend/action/actions/structure_level_list_of_speakers/update.py:47]: You could fetch the `db_instance` once at the start to deduplicate and streamline the code

**Review by jsangmeister — CHANGES_REQUESTED** [review]

**Review by jsangmeister — APPROVED** [review]

**Comment by jsangmeister** [comment]
@Elblinator please test if this fixes your countdown issues in the client

**Comment by luisa-beerboom** [comment]
> If the `remaining_time` is updated while `current_start_time` is not `None` (meaning some speaker is currently speaking for the structure level), the difference between the old and the new remaining time must be added/substracted from the `current_start_time`

After speaking with @Elblinator this change is to be reverted

**Review by Elblinator — APPROVED** [review]
LGTM, it's awesome! 

But please fix your tests :)

**Review by jsangmeister — APPROVED** [review]

**Pull request merged**